src/js/structs/CompositeState.js

```diff
@@ -47,6 +47,7 @@ const mergeObjectsById = function(newData, data = []) {
 class CompositeState {
   constructor(data = {}) {
     this.data = data;
+    this.updateNodesList();
```

Comment:
hm. this could be archived via `setter` instead of manually calling it everytime its changed… 🤔(should be less error prone)